File: app/channels/user_channel.rb
# frozen_string_literal: true

class UserChannel < ApplicationCable::Channel
  def subscribed
    user = User.find(params[:id])
    stream_from "user:#{user.id}"

    transmit(
      { type: "user_info", id: user.id, email: user.email, role: user.role }
    )
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end

File: app/javascript/channels/user_channel.js
import consumer from "channels/consumer";

consumer.subscriptions.create(
  { channel: "UserChannel", id: 1 },
  {
    connected() {
      console.log("✅ Connected to UserChannel");
    },

    disconnected() {
      console.log("🔌 Disconnected from UserChannel");
    },

    received(data) {
      if (data.type === "user_info") {
        console.log("👤 User info:", data);
      } else {
        console.log("📩 Received:", data);
      }
    },
  },
);
